app/src/main/java/ch/epfl/cs311/wanderwave/model/utils/BeaconGenerator.kt

```diff
@@ -0,0 +1,133 @@
+package ch.epfl.cs311.wanderwave.model.utils
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Log
+import androidx.core.content.ContextCompat
+import ch.epfl.cs311.wanderwave.BuildConfig.MAPS_API_KEY
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.data.Location
+import com.google.android.gms.common.api.ApiException
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.api.net.FindCurrentPlaceRequest
+
+// TODO: What would be a good value for these ?
+private const val MIN_REVIEWS = 100
+private const val MIN_BEACON_DISTANCE = 100.0
+
+// TODO: Store as a resource
+// TODO: Which types do we want ?
+val types: List<String> =
+    listOf(
+        "airport",
+        "amusement_park",
+        "aquarium",
+        "art_gallery",
+        "bowling_alley",
+        "campground",
+        "casino",
+        "church",
+        "city_hall",
+        "courthouse",
+        "embassy",
+        "fire_station",
+        "hospital",
+        "library",
+        "light_rail_station",
+        "local_government_office",
+        "movie_theater",
+        "museum",
+        "park",
+        "stadium",
+        "subway_station",
+        "synagogue",
+        "tourist_attraction",
+        "train_station",
+        "transit_station",
+        "university",
+        "zoo")
+
+fun createNearbyBeacons(
+    location: Location,
+    nearbyBeacons: List<Beacon>,
+    radius: Double,
+    context: Context
+): List<Beacon> {
+  // Initialize the list of new Beacons
+  val newBeacons = mutableListOf<Beacon>()
+
+  // Get the nearby points of interest
+  val nearbyPOIs = getNearbyPOIs(context, location, radius)
+
+  // Place new beacons at the nearby points of interest
+  for (poi in nearbyPOIs) {
+    // Check if POI is far enough from existing beacons
+    if (nearbyBeacons.any() { beacon ->
+      beacon.location.distanceBetween(poi) > MIN_BEACON_DISTANCE
+    }) {
+      newBeacons.add(Beacon("", poi))
```

Comment:
For now it's fine, but when @KlaKalma will have prepare the corresponding method, we should look back at this for the id